### ISSUE TITLE
Unhandled exception

### ISSUE THREAD

**Issue by cctgapp** [djkaty/Il2CppInspector#98]
Hi Kat,
  I met a problem when processing this file using latest code. Please help to check it. Thanks!
[download](https://tsehome-my.sharepoint.com/:u:/g/personal/cct_tsehome_onmicrosoft_com/EUr_2hkrGvhMkJtQHgmuQ-gBun_TaNIQGEuH1EgobMe5RA?e=1fQETa)

`Il2CppInspector Command-Line Edition
Version 2020.2
(c) 2017-2020 Katy Coe - www.djkaty.com - www.github.com/djkaty

Detected metadata version 24.1
Container format: Mach-O 64-bit
Container endianness: Little
Architecture word size: 64-bit
Instruction set: ARM64
Global offset: 0x0000000100000000
Unknown symbol type: 2c   0000000005614542   radr://5614542
Symbol table(s) found with 5854 entries
No matches in symbol table
Required structures acquired from code heuristics. Initialization function: 0x00000001003317D0
CodeRegistration struct found at 0x0000000107602850 (file offset 0x07602850)
MetadataRegistration struct found at 0x00000001077333E8 (file offset 0x077333E8)
Unhandled exception. System.ArgumentException: An item with the same key has already been added. Key: 18446744069414584320
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Il2CppInspector.Il2CppBinary.Configure(UInt64 codeRegistration, UInt64 metadataRegistration) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.Common/IL2CPP/Il2CppBinary.cs:line 274
   at Il2CppInspector.Il2CppBinary.InitializeWithoutMetadata(Double version) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.Common/IL2CPP/Il2CppBinary.cs:line 191
   at Il2CppInspector.Il2CppBinary.Initialize(Metadata metadata) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.Common/IL2CPP/Il2CppBinary.cs:line 130
   at Il2CppInspector.Il2CppInspector.LoadFromStream(Stream binaryStream, Stream metadataStream, Boolean silent) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.Common/IL2CPP/Il2CppInspector.cs:line 556
   at Il2CppInspector.Il2CppInspector.LoadFromFile(String binaryFile, String metadataFile, Boolean silent) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.Common/IL2CPP/Il2CppInspector.cs:line 505
   at Il2CppInspector.CLI.App.Run(Options options) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.CLI/Program.cs:line 196
   at Il2CppInspector.CLI.App.<>c.<Main>b__2_0(Options options) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.CLI/Program.cs:line 115
   at CommandLine.ParserResultExtensions.MapResult[TSource,TResult](ParserResult`1 result, Func`2 parsedFunc, Func`2 notParsedFunc)
   at Il2CppInspector.CLI.App.Main(String[] args) in /Users/cheng-chungtseng/Developer/Tools/Il2CppInspector/Il2CppInspector.CLI/Program.cs:line 114

`

**Comment by Toshiba1127** [comment]
I got the same issue but I used the GUI version

IOS Binary and metadata: https://anonymousfiles.io/DQ7sXE70/

![image](https://user-images.githubusercontent.com/64008639/97996860-91c29a00-1de8-11eb-8a06-8df0525dfb1f.png)

**Comment by cctgapp** [comment]
Hi Kat,
  Previous data seems encrypted. I decrypted the file and the result is still the same. I upload them [here](https://www.dropbox.com/s/ht0eoq12l1mxv6o/wr102.zip?dl=0). Please help to check it. Thanks !!

**Comment by djkaty** [comment]
Looked over the files. This is a duplicate of #44 - the fields in Il2CppCodeRegistration and Il2CppMetadataRegistration have been re-arranged a a form of obfuscation.

Automatically determining which fields are which is not a currently supported scenario, however I have added it to the roadmap for a future enhancement.

In the meantime, I have added some code to auto-detect this and give a more descriptive error. You can resolve the problem yourself for individual binaries if you can determine the correct field order for these structs using a disassembler, change the structs in Il2CppBinaryClasses.cs to be correct for your particular binary file, and re-compiling.

**Comment by djkaty** [comment]
This issue is now resolved (for all 3 binaries) in the latest series of commits.

There was also string encryption which I have also resolved in https://github.com/djkaty/Il2CppInspector/commit/12e94921615daf9c135fd48581665b1076fa5e93, https://github.com/djkaty/Il2CppInspector/commit/47ff61a996cd5de4e480d2560661dd5983be2c0c and https://github.com/djkaty/Il2CppInspector/commit/3556697939dce659b309a6491659196d27911110.

The first two files in this thread can also be handled by Il2CppInspector but there was a small bug which caused a crash, this is resolved in https://github.com/djkaty/Il2CppInspector/commit/3ddf04eafccfb3340c89728c77313515fe3231c8.

The difference in output between the first two files and the third attached file in the thread are just that some of the type names etc. are obfuscated in the first two.
